docs/source/deploy.rst

```diff
@@ -0,0 +1,151 @@
+
+       git clone git@github.com:QuantStack/voila.git
+       cd voila/notebooks/
+
+2. Add a requirements.txt file to the project directory. This file should 
+   contain all the Python dependencies your voila app needs to run. For this
+   tutorial we will copy the contents of the environment.yml of voila.
+   We omit xleaflet and xeus-cling because these require extra work that is
+   beyond the scope of this guide.
+
+   .. code:: txt
+
+       ipyvolume
+       bqplot
+       scipy
+       ipympl
+       voila==0.1.2
+       ipympl
+
+Cloud Service Providers
+=======================
+
+Deployment on Heroku
+--------------------
+
+Heroku.com is an attractive option if you want to try out deployment for
+free. You have limited computing hours, however the app will also
+automatically shutdown if it is idle.
+
+The general steps for deployment at Heroku can be found
+`here <https://devcenter.heroku.com/articles/getting-started-with-python>`__.
+High level instructions, specific to voila can be found below:
+
+1. Follow the steps of the official documentation to install the heroku
+   cli and login on your machine.
+2. Add a file named runtime.txt to the project directory with the following
+   content:
+
+   .. code:: txt
+
+       python-3.7.3
+
+3. Add a file named Procfile to the project directory with the
+   following content if you want to show all notebooks:
+
+   ::
+
+       web: voila —-port=$PORT --no-browser
+
+   Or the following if you only want to show one notebook:
+
+   ::
+
+       web: voila —-port=$PORT —-no-browser your_notebook.ipynb
+
+4. Initialize your git repo and commit your code. At minimum you need to commit
+   your notebooks, requirements.txt, runtime.txt, and the Procfile.
+
+   .. code:: bash
+
+       git init
+       git add <your-files>
+       git commit -m "my message"
+
+5. Create an Heroku instance and push the code.
+
+   .. code:: bash
+
+       heroku create
+       git push heroku master
+
+6. Open your web app
+
+   .. code:: bash
+
+       heroku open
+
+
+
+To resolve issues, it is useful to see the logs of your application. You can do this by running:
+
+   .. code:: bash
+
+       heroku logs --tail
+
+
+
+Deployment on Google App Engine
+-------------------------------
+
+You can deploy on `Google App
+Engine <https://cloud.google.com/appengine/>`__ in a “flexible”
+environment. This means that the underlying machine will always run.
+This is more expensive than a “standard” environment, which is similar
+to Heroku’s free option. However, Google App Engine’s “standard”
+environment does not support websockets, which is a requirement for
+voila.
+
+The general steps for deployment at Google App Engine can be found
+`here <https://cloud.google.com/appengine/docs/flexible/python/quickstart>`__.
+High level instructions specific to voila can be found below:
+
+1. Follow the “Before you begin steps” from the official documentation
+   to create your: 1) account, 2) project and 3) App Engine app.
```

Comment:
The inlined numbers look a bit odd on the rendered version of the docs.

How about something like: `to create your account, project and the App Engine app.`

Reply:
Adopted your suggestions